model-optimizer/extensions/back/ShapeOfConstFolding.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from mo.back.replacement import BackReplacementPattern
+from mo.graph.graph import Graph, rename_nodes
+from mo.ops.const import Const
+
+
+class ShapeOfConstFolding(BackReplacementPattern):
+    """
+    The transformation folding Const() -> ShapeOf() to Const()
```

Comment:
`The transformation folding Const() -> ShapeOf() to Const()` -> `The transformation folds ShapeOf(Const) -> Const`.